snuba/settings/validation.py

```diff
@@ -36,6 +38,14 @@ def _validate_settings(locals: Mapping[str, Any]) -> None:
         "transactions-subscription-results",
         "sessions-subscription-results",
     }
+    if settings.ENABLE_METRICS_SUBSCRIPTIONS:
```

Comment:
You don't need to check this setting here, you can just add it to the main `topic_names` dictionary. Putting it on this list doesn't really have any effect, just allows those topics to be configured via KAFKA_TOPIC_MAP and KAFKA_BROKER_CONFIG without erroring.

Reply:
Fwiw I'm not sure why the original PR was reverted earlier - it should be harmless

Reply:
I am not entirely sure as well, but it seemed that the consumers kept restarting according to what @fpacifici told me until I reverted that PR so I assumed it must be that its checking for topic names that don't exist in prod somehow?

Reply:
I don't think so. Can you link/paste the actual error message you were seeing to confirm?

AFAIK the issue the other day was actually to do with this PR https://github.com/getsentry/snuba/pull/2222/files

If you pass a commit log topic to the consumer like that does it will automatically start publishing offsets to that topic. Is it possible the metrics commit log didn't exist in prod yet?

Reply:
@lynnagara Do you reckon I should try to deploy this without the feature flag check since these are not production consumers?

Reply:
https://github.com/getsentry/snuba/blob/master/snuba/settings/validation.py#L40-L46
These are the only lines where `topic_names` is used.. you can see there is no need for the actual topics to exist.

Reply:
Let's confirm what happened last week. I don't think we should be putting feature flags around everything "just in case"